lib/ansible/cli/doc.py

```diff
@@ -147,19 +187,23 @@ def _find_all_collection_roles(self, name_filters=None, collection_filter=None):
             roles_dir = os.path.join(path, 'roles')
             if os.path.exists(roles_dir):
                 for entry in os.listdir(roles_dir):
-                    full_path = os.path.join(roles_dir, entry, 'meta', self.ROLE_ARGSPEC_FILE)
-                    if os.path.exists(full_path):
-                        if name_filters is None:
-                            found.add((entry, collname, path))
-                        else:
-                            # Name filters might contain a collection FQCN or not.
-                            for fqcn in name_filters:
-                                if len(fqcn.split('.')) == 3:
-                                    (ns, col, role) = fqcn.split('.')
-                                    if '.'.join([ns, col]) == collname and entry == role:
+
+                    # Role argspecs can be in one of two files in meta subdir
+                    for specfile in (self.ROLE_ARGSPEC_BASE, self.ROLE_META_FILE):
```

Comment:
This needs to be updated to use file extensions.

Reply:
use `C.YAML_FILENAME_EXTENSIONS ` for that